src/Resource/utils.ts

```diff
@@ -336,3 +340,50 @@ export async function deprecateSelfResource(
     throw error;
   }
 }
+
+export async function getIncomingLinks(
+  orgLabel: string,
+  projectLabel: string,
+  selfUrl: string,
+  paginationSettings: PaginationSettings,
+): Promise<PaginatedList<ResourceLink>> {
+  try {
+    const { from, size } = paginationSettings;
+    const view = await getSparqlView(orgLabel, projectLabel);
+    const countQuery = `SELECT (COUNT(?s) AS ?total)  WHERE { ?s ?p <${selfUrl}>}`;
+    const paginatedQuery = `SELECT ?s ?p WHERE { ?s ?p <${selfUrl}>} LIMIT ${size} OFFSET ${from}`;
+    const countResponse: SparqlViewQueryResponse = await view.query(countQuery);
+    const paginatedResponse: SparqlViewQueryResponse = await view.query(
+      paginatedQuery,
+    );
+    if (countResponse.results && paginatedResponse.results) {
+      const total = +countResponse.results.bindings[0].total.value;
+      const queryResults = paginatedResponse.results.bindings;
+      const resourcePromises = queryResults.map(subjectPredicatePair => {
+        return Resource.getSelf(
+          subjectPredicatePair.s.value,
+          orgLabel,
+          projectLabel,
+        );
+      });
+      const resources = await Promise.all(resourcePromises);
```

Comment:
:sweat_smile: this is hardcore. Do we have to fetch every single resources?

Reply:
I'll do it anyways...